components/Mediator.vue

```diff
@@ -324,14 +329,87 @@ export default {
     },
     async loadHints() {
       let hintOptions;
+      let recommendedHandlerOrigins;
       if(this.credentialRequestOptions) {
         // get matching hints from request options
         hintOptions = await navigator.credentialMediator.ui
           .matchCredentialRequest(this.credentialRequestOptions);
+        ({web: {recommendedHandlerOrigins = []}} =
+          this.credentialRequestOptions);
       } else {
         // must be a storage request, get hints that match credential
         hintOptions = await navigator.credentialMediator.ui
           .matchCredential(this.credential);
+        ({options: {recommendedHandlerOrigins = []} = {}} = this.credential);
+      }
+
+      // no available hints, check for recommended options
+      if(hintOptions.length === 0 && Array.isArray(recommendedHandlerOrigins)) {
+        // get relevant types to match against handler
+        let types = [];
+        if(this.credentialRequestOptions) {
+          // types are all capitalized `{web: {Type1, Type2, ..., TypeN}}`
+          types = Object.keys(this.credentialRequestOptions.web)
+            .filter(k => k[0] === k.toUpperCase()[0]);
+        } else {
+          types.push(this.credential.dataType);
+        }
+
+        // maximum of 3 recommended handlers
+        recommendedHandlerOrigins = recommendedHandlerOrigins.slice(0, 3);
+        const jitHints = (await Promise.all(recommendedHandlerOrigins.map(
+          async recommendedOrigin => {
+            if(typeof recommendedOrigin !== 'string') {
+              return;
+            }
+            const {host, origin} = utils.parseUrl(recommendedOrigin);
+            const manifest = (await getWebAppManifest(host)) || {};
+            const name = manifest.name || manifest.short_name || host;
+            if(!(manifest.credential_handler &&
+              manifest.credential_handler.url &&
+              manifest.credential_handler.enabledTypes)) {
+              // manifest does not have credential handler info
```

Comment:
Are sanity checks needed that url is string and enabledTypes is an array of strings?